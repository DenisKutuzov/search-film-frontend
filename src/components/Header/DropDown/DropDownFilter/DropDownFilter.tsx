import {FC} from 'react'
import style from '@/components/Header/DropDown/DropDown.module.scss'
import Link from 'next/link'

export interface GenresProps {
    list: any
}

const DropDownFilter: FC<GenresProps> = ({list}) => {
    return (
        <>
            <div>
                <p style={{color: 'white', marginBottom: '15px'}}>Жанры</p>
                <ul className={style.dropDown__genres}>
                    {list.genres.map((element, idx) => {
                        return (
                            <div key={idx}>
                                <li>
                                    <Link
                                        className={style.dropDown__a}
                                        href={element.href}
                                    >
                                        {element.name}
                                    </Link>
                                </li>
                            </div>
                        )
                    })}
                </ul>
            </div>
            <div className={style.dropDown__countyYarsBlock}>
                <p style={{color: 'white', marginBottom: '15px'}}>Страны</p>
                <ul className={style.dropDown__country}>
                    {list.country.map((element, idx) => {
                        return (
                            <li key={idx}>
                                <Link className={style.dropDown__a} href={element.href}>
                                    {element.name}
                                </Link>
                            </li>
                        )
                    })}
                </ul>
                <p
                    style={{
                        color: 'white',
                        marginTop: '15px',
                        marginBottom: '10px',
                    }}
                >
                    Годы
                </p>
                <ul className={style.dropDown__country}>
                    {list.years.map((element, idx) => {
                        return (
                            <li key={idx}>
                                <Link className={style.dropDown__a} href={element.href}>
                                    {element.name}
                                </Link>
                            </li>
                        )
                    })}
                </ul>
            </div>
        </>
    )
}

export default DropDownFilter