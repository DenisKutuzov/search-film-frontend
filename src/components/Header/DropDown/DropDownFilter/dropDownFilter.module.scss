.dropDown__genres {
  display: flex;
  flex-flow: column wrap;
  gap: 10px;
  //width: 300px;
  height: 330px;
  width: 400px;
}

.dropDown__countryYearsBlock {
  display: flex;
  flex-direction: column;
  width: 300px;
}


.dropDown__country {
  display: flex;
  flex-direction: column;
  gap: 10px;
}

.dropDown__a {
  font-size: 16px;
  font-weight: 400;
  color: rgba(255, 255, 255, .48);
  transition-duration: .4s;

  &:hover {
    color: white;
  }
}