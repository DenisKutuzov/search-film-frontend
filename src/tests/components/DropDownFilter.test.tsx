import DropDownFilter from "@/components/Header/DropDown/DropDownFilter/DropDownFilter";
import {  render, screen } from "@testing-library/react";

describe("DropDownFilter", () => {
    it("DropDownFilter renders without crashing ", () => {
        const list = {
            genres: [{ nameRU: 'тестовый жанр', href: 'ссылка' }],
            country: [{ name: 'тестовая страна', href: 'ссылка' }],
            years: [{ name: 'тестовый год', href: 'ссылка' }]
        }

        const component = render(<DropDownFilter list={list} />);
        expect(component).toMatchSnapshot();
        const lists = screen.getAllByRole('list')
        expect(lists).toHaveLength(3);
        const testGenre = screen.getByText('тестовый жанр')
        const testCountry = screen.getByText('тестовая страна')
        const testYear = screen.getByText('тестовый год')
        expect(testGenre).toBeInTheDocument();
        expect(testCountry).toBeInTheDocument();
        expect(testYear).toBeInTheDocument();

    });
});
