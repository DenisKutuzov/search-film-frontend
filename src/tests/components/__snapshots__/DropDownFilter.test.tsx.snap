// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDownFilter DropDownFilter renders without crashing  1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <p
          style="color: white; margin-bottom: 15px;"
        >
          Жанры
        </p>
        <ul
          class="dropDown__genres"
        >
          <div>
            <li>
              <a
                class="dropDown__a"
                href="/movies/undefined"
              >
                тестовый жанр
              </a>
            </li>
          </div>
        </ul>
      </div>
      <div
        class="dropDown__countryYearsBlock"
      >
        <p
          style="color: white; margin-bottom: 15px;"
        >
          Страны
        </p>
        <ul
          class="dropDown__country"
        >
          <li>
            <a
              class="dropDown__a"
              href="ссылка"
            >
              тестовая страна
            </a>
          </li>
        </ul>
        <p
          style="color: white; margin-top: 15px; margin-bottom: 10px;"
        >
          Годы
        </p>
        <ul
          class="dropDown__country"
        >
          <li>
            <a
              class="dropDown__a"
              href="ссылка"
            >
              тестовый год
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <p
        style="color: white; margin-bottom: 15px;"
      >
        Жанры
      </p>
      <ul
        class="dropDown__genres"
      >
        <div>
          <li>
            <a
              class="dropDown__a"
              href="/movies/undefined"
            >
              тестовый жанр
            </a>
          </li>
        </div>
      </ul>
    </div>
    <div
      class="dropDown__countryYearsBlock"
    >
      <p
        style="color: white; margin-bottom: 15px;"
      >
        Страны
      </p>
      <ul
        class="dropDown__country"
      >
        <li>
          <a
            class="dropDown__a"
            href="ссылка"
          >
            тестовая страна
          </a>
        </li>
      </ul>
      <p
        style="color: white; margin-top: 15px; margin-bottom: 10px;"
      >
        Годы
      </p>
      <ul
        class="dropDown__country"
      >
        <li>
          <a
            class="dropDown__a"
            href="ссылка"
          >
            тестовый год
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
